R/clean_variable_spelling.R

```diff
@@ -211,9 +234,52 @@ clean_variable_spelling <- function(x = data.frame(), wordlists = list(), spelli
     wordlists    <- wordlists[names(wordlists) != ".global"]
     has_global   <- !is.null(global_words)
     # Iterate over the names of the dictionaries -----------
-    to_iterate <- intersect(names(wordlists), names(x))
+    
+    
+    # extract vars to check from wordlists (both plain and regex)
+    vars_check <- names(wordlists)
+    is_var_regex <- grepl("^\\.regex", vars_check)
```

Comment:
Two things:

1. At this point, I think it would be good to have a control structure to check if any of the vars contain the regex keyword. If the regex isn't present, then there's no need to perform any of this manipulation.
2. This grep could match a column called .regexam, so I'm suggesting an explicit space here
```suggestion
    is_var_regex <- grepl("^\\.regex[[:space:]]", vars_check)
```